test/e2e/clusterctl_upgrade_test.go

```diff
@@ -62,13 +69,13 @@ var _ = Describe("When testing clusterctl upgrades (v1.4=>current)", func() {
 			ArtifactFolder:         artifactFolder,
 			SkipCleanup:            skipCleanup,
 			InfrastructureProvider: pointer.String("docker"),
-			InitWithBinary:         "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.5/clusterctl-{OS}-{ARCH}",
+			InitWithBinary:         "https://github.com/kubernetes-sigs/cluster-api/releases/download/v" + stableRelease + "/clusterctl-{OS}-{ARCH}",
```

Comment:
Let's add some const's and use fmt.Sprintf instead.

```go
var clusterctlDownloadURL = "https://github.com/kubernetes-sigs/cluster-api/releases/download/v%s/clusterctl-{OS}-{ARCH}"
```

```suggestion
			InitWithBinary:         fmt.Sprintf(clusterctlDownloadURL, stableRelease),
```

Also propably for the other strings

Reply:
fixed, and also did some further refactoring. please check.